app/controllers/users/book_requests_controller.rb

```diff
@@ -8,13 +11,67 @@ def index
     end
 
     def edit
-      @book_request = BookRequest.find_by(params[:id])
+      return if current_user&.id == @book_request&.user_id
+
+      handle_danger_flash
+    end
+
+    def update
+      @book_request = BookRequest.find(params[:id])
+      if @book_request.update_attributes(book_request_params)
+        destroy_image_files unless image_file_params
+        create_image_files
+        flash[:success] = t('book_requests.update.success')
+        redirect_to user_book_request_path(current_user, @book_request)
+      else
+        render 'edit'
```

Comment:
should add warning flash

Reply:
and then add in rspec for this case

Reply:
warning ready in edit template(from shared/error_message)